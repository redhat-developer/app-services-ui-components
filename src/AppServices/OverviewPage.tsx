import {
  Button,
  ButtonVariant,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderMain,
  CardTitle,
  Grid,
  Label,
  Stack,
  StackItem,
  Title,
} from "@patternfly/react-core";
import { ExternalLinkAltIcon } from "@patternfly/react-icons";
import type { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";
import {
  AppServicesOverviewIconPattern,
  LogoRedHatOpenShiftApiManagementAStandardRgb,
  LogoRedHatOpenShiftDatabaseAccessAStandardRgb,
  LogoRedHatOpenShiftDataScienceAStandardRgb,
  LogoRedHatOpenShiftServiceRegistryAStandardRgb,
  LogoRedHatOpenShiftStreamsForApacheKafkaAStandardRgbPng,
} from "../images";
import {
  MarketingPageBanner,
  MarketingPageHero,
  MarketingPageSection,
} from "./components";

export type OverviewPageProps = {
  toKafkaHref: string;
  toServiceRegistryHref: string;
};

export const OverviewPage: FunctionComponent<OverviewPageProps> = ({
  toKafkaHref,
  toServiceRegistryHref,
}) => {
  const { t } = useTranslation();
  return (
    <>
      <MarketingPageHero
        title={t("overview:heroTitle")}
        tagLine={t("overview:heroTagline")}
        description={t("overview:heroDescription")}
        description2={t("overview:heroDescription2")}
        heroImage={AppServicesOverviewIconPattern}
        heroImageSize={678}
        cta={
          <Button
            variant={ButtonVariant.primary}
            ouiaId="button-try-kafka"
            isLarge
            component={(props) => (
              <Link
                {...props}
                data-testid="hero-buttonTryKafka"
                to={toKafkaHref}
              />
            )}
          >
            {t("overview:heroCallToActionButton")}
          </Button>
        }
        variant="dark"
      />
      <MarketingPageBanner>{t("overview:banner")}</MarketingPageBanner>
      <MarketingPageSection>
        <Grid md={6} lg={3} hasGutter>
          {/* API Management card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftApiManagementAStandardRgb}
                  alt="Red Hat OpenShift API Management logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview:rhoamTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem>{t("overview:rhoamMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview:rhoamSecondaryText")}{" "}
                  <Button
                    data-testid="cardRHOAM-linkOpenShift"
                    isInline
                    variant={ButtonVariant.link}
                    component="a"
                    target="_blank"
                    href="https://openshift.com"
                  >
                    OpenShift
                    <ExternalLinkAltIcon className="pf-u-ml-xs" />
                  </Button>
                  .
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                variant={ButtonVariant.secondary}
                data-testid="cardRHOAM-buttonCTA"
                component="a"
                target="_blank"
                href="https://developers.redhat.com/products/rhoam/getting-started"
              >
                {t("overview:getStarted")}{" "}
                <ExternalLinkAltIcon className="pf-u-ml-sm" />
              </Button>
            </CardFooter>
          </Card>

          {/* Data science card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftDataScienceAStandardRgb}
                  alt="Red Hat OpenShift Data Science logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview:rhodsTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <Label>{t("overview:fieldTrial")}</Label>
                </StackItem>
                <StackItem>{t("overview:rhodsMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview:rhodsSecondaryText")}{" "}
                  <Button
                    data-testid="cardRHODS-linkOpenShift"
                    isInline
                    variant={ButtonVariant.link}
                    component="a"
                    target="_blank"
                    href="https://openshift.com"
                  >
                    {" "}
                    OpenShift
                    <ExternalLinkAltIcon className="pf-u-ml-xs" />
                  </Button>
                  .
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHODS-buttonTryIt"
                ouiaId="button-rhods-tryit"
                variant={ButtonVariant.secondary}
                component="a"
                target="_blank"
              >
                {t("overview:tryIt")}
              </Button>
            </CardFooter>
          </Card>

          {/* Service Registry card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftServiceRegistryAStandardRgb}
                  alt="Red Hat OpenShift Service Registry logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview:rhosrTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem>{t("overview:rhosrMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview:rhosrSecondaryText")}
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHOSR-buttonCTA"
                variant={ButtonVariant.secondary}
                component={(props) => (
                  <Link
                    {...props}
                    data-testid="hero-buttonTryKafka"
                    to={toServiceRegistryHref}
                  />
                )}
              >
                {t("overview:rhosrCallToActionButton")}
              </Button>
            </CardFooter>
          </Card>

          {/* Kafka card */}
          <Card ouiaId="card-rhosak">
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftStreamsForApacheKafkaAStandardRgbPng}
                  alt="Red Hat OpenShift Streams for Apache Kafka logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview:rhosakTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem>{t("overview:rhosakMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview:rhosakSecondaryText")}
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHOSAK-buttonCreateKafka"
                variant={ButtonVariant.secondary}
                component={(props) => <Link {...props} to={toKafkaHref} />}
                ouiaId="button-create"
              >
                {t("overview:rhosakCallToActionButton")}
              </Button>
            </CardFooter>
          </Card>

          {/* Database access card */}
          <Card>
            <CardHeader>
              <CardHeaderMain>
                <img
                  src={LogoRedHatOpenShiftDatabaseAccessAStandardRgb}
                  alt="Red Hat OpenShift Data Science logo"
                  style={{ height: "50px" }}
                />
              </CardHeaderMain>
            </CardHeader>
            <CardTitle>
              <Title headingLevel="h2" size="xl">
                {t("overview:dbaasTitle")}
              </Title>
            </CardTitle>
            <CardBody>
              <Stack hasGutter>
                <StackItem className="pf-u-mb-lg">
                  <Label>{t("overview:servicePreview")}</Label>
                </StackItem>
                <StackItem>{t("overview:dbaasMainText")}</StackItem>
                <StackItem className="pf-u-color-200">
                  {t("overview:dbaasSecondaryText")}{" "}
                  <Button
                    data-testid="cardRHODS-linkOpenShift"
                    isInline
                    variant={ButtonVariant.link}
                    component="a"
                    target="_blank"
                    href="https://openshift.com"
                  >
                    {" "}
                    OpenShift
                    <ExternalLinkAltIcon className="pf-u-ml-xs" />
                  </Button>
                  .
                </StackItem>
              </Stack>
            </CardBody>
            <CardFooter>
              <Button
                data-testid="cardRHODS-buttonCTA"
                variant={ButtonVariant.secondary}
                component="a"
                target="_blank"
                href="https://console.redhat.com/application-services/databases"
              >
                {t("overview:learnMore")}{" "}
                <ExternalLinkAltIcon className="pf-u-ml-sm" />
              </Button>
            </CardFooter>
          </Card>
        </Grid>
      </MarketingPageSection>
    </>
  );
};
