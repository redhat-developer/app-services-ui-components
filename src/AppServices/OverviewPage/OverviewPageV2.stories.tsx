import { ComponentMeta, ComponentStory } from "@storybook/react";
import { OverviewPageV2 } from "./OverviewPageV2";

export default {
  component: OverviewPageV2,
  args: {
    toKafkaHref: "/to-kafka",
    toServiceRegistryHref: "/to-service",
    toConnectorsHref: "/to-connectors",
  },
} as ComponentMeta<typeof OverviewPageV2>;

const Template: ComponentStory<typeof OverviewPageV2> = (args) => (
  <OverviewPageV2 {...args} />
);

export const Example = Template.bind({});
