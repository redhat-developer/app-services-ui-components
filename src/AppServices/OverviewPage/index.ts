export * from "./OverviewPage";
export * from "./OverviewPageV2";
