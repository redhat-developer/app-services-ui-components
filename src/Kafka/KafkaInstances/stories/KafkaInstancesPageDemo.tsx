import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
