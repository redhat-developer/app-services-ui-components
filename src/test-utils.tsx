import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
