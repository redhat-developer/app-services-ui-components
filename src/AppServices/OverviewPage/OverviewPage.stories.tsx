import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { OverviewPage } from "./OverviewPage";

export default {
  component: OverviewPage,
  args: {
    toKafkaHref: "/to-kafka",
    toServiceRegistryHref: "/to-service",
  },
} as ComponentMeta<typeof OverviewPage>;

const Template: ComponentStory<typeof OverviewPage> = (args) => (
  <OverviewPage {...args} />
);

export const Example = Template.bind({});
