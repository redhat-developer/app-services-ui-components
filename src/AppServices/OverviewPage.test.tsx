import { composeStories } from "@storybook/testing-react";
import { render, waitForI18n } from "../test-utils";
import * as stories from "./OverviewPage.stories";

const { OverviewPage } = composeStories(stories);

describe("OverviewPage", () => {
  it("renders", async () => {
    const comp = render(<OverviewPage />);
    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
    expect(
      comp.getByText("Try OpenShift Streams for Apache Kafka")
    ).toHaveAttribute("href", OverviewPage.args!.toKafkaHref);
    const [createServiceRegistry, createKafka] =
      comp.getAllByText("Create an instance");
    expect(createKafka).toHaveAttribute("href", OverviewPage.args!.toKafkaHref);
    expect(createServiceRegistry).toHaveAttribute(
      "href",
      OverviewPage.args!.toServiceRegistryHref
    );

    expect(comp.getByTestId("cardRHODS-buttonTryIt")).toHaveAttribute(
      "data-ouia-component-id",
      "button-rhods-tryit"
    );

    expect(
      comp.getByText("Try OpenShift Streams for Apache Kafka")
    ).toHaveAttribute("data-ouia-component-id", "button-try-kafka");
  });
});
