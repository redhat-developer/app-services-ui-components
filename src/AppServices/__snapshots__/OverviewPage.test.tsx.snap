// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewPage renders 1`] = `
<body>
  <div>
    <section
      class="pf-c-page__main-section pf-m-dark-200 appsrv-marketing--banner pf-u-background-color-dark-100"
      style="--appsrv-marketing--banner--before--BackgroundImage: url(dummyimage); --appsrv-marketing--banner--before--BackgroundSize: 678px; --appsrv-marketing--banner--before--BackgroundRepeat: repeat; --appsrv-marketing--banner--before--BackgroundPositionY: 0px;"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <h1
          class="pf-c-title pf-m-2xl"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Get started with Red Hat OpenShift Application Services
        </h1>
        <p
          class="appsrv-marketing--banner__tagline "
          data-ouia-component-id="OUIA-Generated-Text-1"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Cloud services for managed OpenShift
        </p>
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-2"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Red Hat OpenShift Application Services deliver a streamlined developer experience for building,deploying, and scaling cloud-native applications.
        </p>
        <p
          class="pf-u-mt-md"
          data-ouia-component-id="OUIA-Generated-Text-3"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Don't miss your chance to preview one of our new application services.
        </p>
        <div
          class="pf-l-stack__item"
        >
          <a
            aria-disabled="false"
            class="pf-c-button pf-m-primary pf-m-display-lg"
            data-ouia-component-id="button-try-kafka"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            data-testid="hero-buttonTryKafka"
            href="/to-kafka"
          >
            Try OpenShift Streams for Apache Kafka
          </a>
        </div>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-light appsrv-marketing--page-section--marketing pf-u-background-color-100"
    >
      <div
        class="pf-l-bullseye"
      >
        <p
          class="appsrv-marketing--banner--text-only"
          data-ouia-component-id="OUIA-Generated-Text-4"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Reduce operational costs and complexity for application development teams with hosted and managed OpenShift Application Services.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width appsrv-marketing--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-l-grid pf-m-all-6-col-on-md pf-m-all-3-col-on-lg pf-m-gutter"
        >
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-1"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <img
                  alt="Red Hat OpenShift API Management logo"
                  src="dummyimage"
                  style="height: 50px;"
                />
              </div>
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-2"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Red Hat OpenShift API Management
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item"
                >
                  API Management supports application development teams that want to take an API-first approach to building microservices-based applications.
                </div>
                <div
                  class="pf-l-stack__item pf-u-color-200"
                >
                  This service is hosted and managed by Red Hat as an add-on for managed
                   
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-ouia-component-id="OUIA-Generated-Button-link-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardRHOAM-linkOpenShift"
                    href="https://openshift.com"
                    target="_blank"
                  >
                    OpenShift
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-xs"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                  .
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="OUIA-Generated-Button-secondary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardRHOAM-buttonCTA"
                href="https://developers.redhat.com/products/rhoam/getting-started"
                target="_blank"
              >
                Get started
                 
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-sm"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-2"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <img
                  alt="Red Hat OpenShift Data Science logo"
                  src="dummyimage"
                  style="height: 50px;"
                />
              </div>
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-3"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Red Hat OpenShift Data Science
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item pf-u-mb-lg"
                >
                  <span
                    class="pf-c-label"
                  >
                    <span
                      class="pf-c-label__content"
                    >
                      Field trial
                    </span>
                  </span>
                </div>
                <div
                  class="pf-l-stack__item"
                >
                  Data Science provides a fully-supported sandbox to rapidly develop, train, and test containerized machine learning models in the public cloud before deploying in production.
                </div>
                <div
                  class="pf-l-stack__item pf-u-color-200"
                >
                  This service is hosted and managed by Red Hat as an add-on for managed
                   
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-ouia-component-id="OUIA-Generated-Button-link-2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardRHODS-linkOpenShift"
                    href="https://openshift.com"
                    target="_blank"
                  >
                     
                    OpenShift
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-xs"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                  .
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="button-rhods-tryit"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardRHODS-buttonTryIt"
                target="_blank"
              >
                Try it
              </a>
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-3"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <img
                  alt="Red Hat OpenShift Service Registry logo"
                  src="dummyimage"
                  style="height: 50px;"
                />
              </div>
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-4"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Red Hat OpenShift Service Registry
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item"
                >
                  Service Registry provides a metadata layer for developers and applications to publish, discover and evolve schema definitions and API designs.
                </div>
                <div
                  class="pf-l-stack__item pf-u-color-200"
                >
                  This service is fully hosted and managed by Red Hat and offered at no cost to Streams for Apache Kafka and API Management users.
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="OUIA-Generated-Button-secondary-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="hero-buttonTryKafka"
                href="/to-service"
              >
                Create an instance
              </a>
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="card-rhosak"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <img
                  alt="Red Hat OpenShift Streams for Apache Kafka logo"
                  src="dummyimage"
                  style="height: 50px;"
                />
              </div>
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-5"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Red Hat OpenShift Streams for Apache Kafka
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item"
                >
                  Streams for Apache Kafka enables IT development teams to capture, process, and stream real-time data across hybrid cloud environments.
                </div>
                <div
                  class="pf-l-stack__item pf-u-color-200"
                >
                  This service is fully hosted and managed by Red Hat for stream-based applications
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="button-create"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardRHOSAK-buttonCreateKafka"
                href="/to-kafka"
              >
                Create an instance
              </a>
            </div>
          </article>
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-4"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <img
                  alt="Red Hat OpenShift Data Science logo"
                  src="dummyimage"
                  style="height: 50px;"
                />
              </div>
            </div>
            <div
              class="pf-c-card__title"
            >
              <h2
                class="pf-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-6"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Red Hat OpenShift Database Access
              </h2>
            </div>
            <div
              class="pf-c-card__body"
            >
              <div
                class="pf-l-stack pf-m-gutter"
              >
                <div
                  class="pf-l-stack__item pf-u-mb-lg"
                >
                  <span
                    class="pf-c-label"
                  >
                    <span
                      class="pf-c-label__content"
                    >
                      Service Preview
                    </span>
                  </span>
                </div>
                <div
                  class="pf-l-stack__item"
                >
                  OpenShift Database Access helps developers find, connect, and provision cloud database services such as MongoDB Atlas, Crunchy Bridge and CockroachDB.
                </div>
                <div
                  class="pf-l-stack__item pf-u-color-200"
                >
                  This service is hosted and managed by Red Hat as an add-on for managed
                   
                  <a
                    aria-disabled="false"
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-ouia-component-id="OUIA-Generated-Button-link-3"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="cardRHODS-linkOpenShift"
                    href="https://openshift.com"
                    target="_blank"
                  >
                     
                    OpenShift
                    <svg
                      aria-hidden="true"
                      class="pf-u-ml-xs"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                      />
                    </svg>
                  </a>
                  .
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-ouia-component-id="OUIA-Generated-Button-secondary-3"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardRHODS-buttonCTA"
                href="https://console.redhat.com/application-services/databases"
                target="_blank"
              >
                Learn more
                 
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-sm"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </article>
        </div>
      </div>
    </section>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
